C7/Util.cs

```diff
@@ -107,11 +107,83 @@ static public Pcx LoadPCX(string relPath)
 		if (PcxCache.ContainsKey(relPath)) {
 			return PcxCache[relPath];
 		}
-		Pcx thePcx = new Pcx(Util.Civ3MediaPath(relPath));
+		Pcx thePcx = new Pcx(Civ3MediaPath(relPath));
 		PcxCache[relPath] = thePcx;
 		return thePcx;
 	}
 
+	// Creates a texture from raw palette data. The data must be 256 pixels by 3 channels. Returns a 16x16 unfiltered RGB texture.
+	public static ImageTexture createPaletteTexture(byte[,] raw)
+	{
+		if ((raw.GetLength(0) != 256) || (raw.GetLength(1) != 3))
+			throw new Exception("Invalid palette dimensions. Palettes must be 256x3.");
+
+		// Flatten palette data since CreateFromData can't accept two-dimensional arrays
+		byte[] flatPalette = new byte[3*256];
+		for (int n = 0; n < 256; n++)
+			for (int k = 0; k < 3; k++)
+				flatPalette[k + 3 * n] = raw[n, k];
+
+		var img = new Image();
+		img.CreateFromData(16, 16, false, Image.Format.Rgb8, flatPalette);
+		var tex = new ImageTexture();
+		tex.CreateFromImage(img, 0);
+		return tex;
+	}
+
+	// Creates textures from a PCX file without de-palettizing it. Returns two ImageTextures, the first is 16x16 with RGB8 format containing the
+	// color palette and the second is the size of the image itself and contains the indices in R8 format.
+	public static (ImageTexture palette, ImageTexture indices) loadPalettizedPCX(string filePath)
```

Comment:
In my opinion it makes sense in this case to have two return values since they're two parts loaded from the same file. It might help to rename this function so it's clear that's what it's doing.

About the R8 format, it's for the texture that stores palette indices so we only need one byte per pixel there. Actually it's not ideal but it's the best format Godot offers. What would be ideal would be a format that corresponds to OpenGL's GL_R8 so we could get the palette index values as integers 0 - 255 directly in the shader, but Godot's R8 format corresponds to GL_RED which can only be read as floating point values.